concepts/functions/about.md

```diff
@@ -1,3 +1,302 @@
 # About
 
-TODO: add information on functions concept
+A [`function`][function] is a block of organized, reusable code that is used to perform a single, related action.
+`Functions` provide better modularity for your application and a high degree of code reusing.
+Like other programming languages, python provides _pre-defined functions_ ([`print`][print], [`map`][map] etc) that are readily available in the Python language. 
+However, you can define your own functions as well which are called [`user-defined functions`][user defined functions].
+Functions can be as simple as _printing a message to the console_ or as complex as _performing a calculation_.
+
+To execute the code inside a function, you need to call the function, which is done by using the function name followed by parenthesis [`()`].
+Data known as `parameters` can be passed to the function inside the parenthesis. Broader terms for parameters are `arguments`. Functions can perform different tasks depending on the value of the parameters.
+
+The function can also return a value using the `return` keyword. The value returned by the function is known as the `return value`. The return value is returned to the caller of the function.
+
+## Creation
+
+In python, functions are created using the `def` keyword. The function definition is followed by the function name and parenthesis [`()`]. Inside the parenthesis, the parameters are specified, and are separated by commas. After the parameters, the colon (`:`) is used to separate the function header from the function body.
+
+The function body is a block of code that is executed when the function is called. The body of the function is indented. The indentation is important because the function body is a block of code. A value can be returned from the function by using the `return` keyword, which will be used by the caller of the function.
+
+```python
+def function_name(parameter1, parameter2, ...):
+    # function body
+    return value
+```
+
+We can also define a function without any parameters or return value.
+```python
+def function_name():
+    # function body
+```
+
+
+## Calling a Function
+
+To call a function, use the function name followed by parenthesis [`()`]. The parameters passed to the function are separated by commas.
+
+Consider the following function:
+```python
+def wish():
+    print("Hello")
+```
+
+Above function can be called by using the following syntax:
+```python
+>>> wish()
+Hello
+```
+
+## Parameters
+
+Parameters are values that are passed to the function when it is called. They are known as `arguments`.
+
+Let's define a function `add`:
+```python
+def add(x, y):
+    print(x + y)
+```
+
+When the function is called, the parameters are passed to the function. We need to pass values for both the parameters, otherwise [`TypeError`][type-error] will be raised.
```

Comment:
```suggestion
When the function is called, the parameters are passed to the function. We need to pass values for both the parameters, otherwise a [`TypeError`][type-error] will be raised.
```